### PR TITLE
kie-issues#961: Adapt kogito-examples - replace image pointers and removing devui references

### PR DESCRIPTION
**Pull request by fantonangeli**
Closes https://github.com/apache/incubator-kie-issues/issues/961

**Notes:**
I created a spreadsheet for future memory about these changes:
https://docs.google.com/spreadsheets/d/1cLLJyORAbUxCbSXEJRSrk-jw3gTS1c5TdYTkqBGNJCw/edit?usp=sharing

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

**WARNING! Please make sure you are opening your PR against `main` branch!**

- [ ] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request checks</b>  
  Please add comment: <b>Jenkins retest this</b>

- for a <b>specific pull request check</b>  
  Please add comment: <b>Jenkins (re)run [kogito-examples] tests</b>

- for <b>quarkus branch checks</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins run quarkus-branch</b>

- for a <b>quarkus branch specific check</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins (re)run [kogito-examples] quarkus-branch</b>

- for <b>quarkus main checks</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins run quarkus-main</b>

- for a <b>specific quarkus main check</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins (re)run [kogito-examples] quarkus-main</b>

- for <b>quarkus lts checks</b>  
  Run checks against Quarkus lts branch  
  Please add comment: <b>Jenkins run quarkus-lts</b>

- for a <b>specific quarkus lts check</b>  
  Run checks against Quarkus lts branch  
  Please add comment: <b>Jenkins (re)run [kogito-examples] quarkus-lts</b>
 
- for <b>native checks</b>  
  Run native checks  
  Please add comment: <b>Jenkins run native</b>

- for a <b>specific native check</b>  
  Run native checks 
  Please add comment: <b>Jenkins (re)run [kogito-examples] native</b>

- for <b>native lts checks</b>  
  Run native checks against quarkus lts branch
  Please add comment: <b>Jenkins run native-lts</b>

- for a <b>specific native lts check</b>  
  Run native checks against quarkus lts branch 
  Please add comment: <b>Jenkins (re)run [kogito-examples] native-lts</b>
 
</details>

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Comment by fantonangeli** [comment]
Hi @paulovmr and @tiagobento,
this PR is ready for review.

File: kogito-springboot-examples/pom.xml [hunk 1]
```diff
@@ -127,7 +125,6 @@
         <module>process-springboot-example</module>
         <module>process-timer-springboot</module>
         <module>process-usertasks-custom-lifecycle-springboot</module>
-        <module>process-usertasks-springboot-with-console</module>
         <module>process-usertasks-springboot</module>
         <module>process-usertasks-with-security-oidc-springboot-with-console</module>
```
Comment (paulovmr) [inline kogito-springboot-examples/pom.xml]: I believe this line should be removed too.

File: kogito-quarkus-examples/pom.xml [hunk 2]
```diff
@@ -95,11 +95,9 @@
         <module>process-service-calls-quarkus</module>
         <module>process-timer-quarkus</module>
         <module>process-usertasks-custom-lifecycle-quarkus</module>
-        <module>process-usertasks-quarkus-with-console</module>
```
Comment (paulovmr) [inline kogito-quarkus-examples/pom.xml:98]: Please consider removing this module from the other profile as well.

File: kogito-quarkus-examples/pom.xml [hunk 3]
```diff
@@ -95,11 +95,9 @@
...
         <module>process-usertasks-quarkus</module>
         <module>process-usertasks-timer-quarkus-with-console</module>
         <module>process-usertasks-timer-data-index-persistence-addon-quarkus</module>
-        <module>process-usertasks-with-security-oidc-quarkus-with-console</module>
```
Comment (paulovmr) [inline kogito-quarkus-examples/pom.xml:102]: Please consider removing this module from the other profile as well.

File: serverless-workflow-examples/pom.xml [hunk 4]
```diff
@@ -47,7 +47,6 @@
         <module>serverless-workflow-callback-quarkus</module>
         <module>serverless-workflow-camel-routes</module>
         <module>serverless-workflow-compensation-quarkus</module>
-        <module>serverless-workflow-consuming-events-over-http-quarkus</module>
```
Comment (paulovmr) [inline serverless-workflow-examples/pom.xml]: I believe we can keep this module and just remove the Dev UI references from it. The README has some steps using `curl` as well, which are still valid. The section `Running the example from Quarkus Dev UI` could be removed.

**Review by paulovmr — CHANGES_REQUESTED** [review]
Hi @fantonangeli , thanks for the PR! I left a few comments inline.

**Review by paulovmr — APPROVED** [review]

**Comment by tiagobento** [comment]
@fantonangeli Can we re-run the CI? Just to be sure.. It's been a while since it ran..

**Review by tiagobento — APPROVED** [review]

**Comment by kie-ci3** [comment]
**PR job** `#3` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>



build-chain build full_downstream  -f 'https://raw.githubusercontent.com/${AUTHOR:apache}/incubator-kie-kogito-pipelines/${BRANCH:main}/.ci/buildchain-config-pr-cdb.yaml' -o 'bc' -p apache/incubator-kie-kogito-examples -u https://github.com/apache/incubator-kie-kogito-examples/pull/1900 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-examples-pr/job/PR-1900/3/display/redirect

**Test results:**
- PASSED: 381
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-examples-pr/job/PR-1900/3/testReport/org.acme/AcmeExchangeResourceIT/unauthorizedTest/">org.acme.AcmeExchangeResourceIT.unauthorizedTest</a></summary>
java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors<br/> [error]: Build step io.quarkus.oidc.deployment.devservices.keycloak.KeycloakDevServicesProcessor#startKeycloakContainer threw an exception: java.lang.RuntimeException: org.testcontainers.containers.ContainerLaunchException: Container startup failed for image quay.io/keycloak/keycloak:22.0.5<br/> at io.quarkus.oidc.deployment.devservices.keycloak.KeycloakDevServicesProcessor.startKeycloakContainer(KeycloakDevServicesProcessor.java:248)<br/> at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)<br/> at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)<br/> at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)<br/> at java.base/java.lang.reflect.Method.invoke(Method.java:568)<br/> at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:864)<br/> at io.quarkus.builder.BuildContext.run(BuildContext.java:282)<br/> at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)<br/> at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)<br/> at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)<br/> at java.base/java.lang.Thread.run(Thread.java:833)<br/> at org.jboss.threads.JBossThread.run(JBossThread.java:501)<br/>Caused by: org.testcontainers.containers.ContainerLaunchException: Container startup failed for image quay.io/keycloak/keycloak:22.0.5<br/> at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:349)<br/> at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:322)<br/> at io.quarkus.oidc.deployment.devservices.keycloak.KeycloakDevServicesProcessor.lambda$startContainer$4(KeycloakDevServicesProcessor.java:374)<br/> at java.base/java.util.Optional.orElseGet(Optional.java:364)<br/> at io.quarkus.oidc.deployment.devservices.keycloak.KeycloakDevServicesProcessor.startContainer(KeycloakDevServicesProcessor.java:399)<br/> at io.quarkus.oidc.deployment.devservices.keycloak.KeycloakDevServicesProcessor.startKeycloakContainer(KeycloakDevServicesProcessor.java:196)<br/> ... 11 more<br/>Caused by: org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception<br/> at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:88)<br/> at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:334)<br/> ... 16 more<br/>Caused by: org.testcontainers.containers.ContainerLaunchException: Could not create/start container<br/> at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:553)<br/> at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:344)<br/> at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)<br/> ... 17 more<br/>Caused by: org.testcontainers.containers.ContainerLaunchException: Timed out waiting for log output matching '.*Keycloak.*started.*'<br/> at org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy.waitUntilReady(LogMessageWaitStrategy.java:47)<br/> at org.testcontainers.containers.wait.strategy.AbstractWaitStrategy.waitUntilReady(AbstractWaitStrategy.java:52)<br/> at org.testcontainers.containers.GenericContainer.waitUntilContainerStarted(GenericContainer.java:964)<br/> at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:490)<br/> ... 19 more<br/>
</details>

**Comment by tiagobento** [comment]
The Jenkins check failed. Should we merge it anyway? Please someone more familiar with the infrastructure here, could you help understand if we can proceed with this or not?

**Comment by fjtirado** [comment]
@tiagobento Merged, the failure was unrelated

**Pull request merged**